app/vmagent/main.go

```diff
@@ -157,6 +158,15 @@ func requestHandler(w http.ResponseWriter, r *http.Request) bool {
 		w.Header().Set("Content-Type", "text/plain")
 		promscrape.WriteHumanReadableTargetsStatus(w)
 		return true
+	case "/-/reload":
+		promscrapeConfigReloadRequests.Inc()
+		if err := syscall.Kill(syscall.Getpid(), syscall.SIGHUP); err != nil {
+			promscrapeConfigReloadErrors.Inc()
+			logger.Errorf("Fail to reload config file, %s", err)
+			return true
```

Comment:
return true, maybe a typo?

Reply:
Thanks~  
It should be a `httpserver.Errorf()` instead of the `logger.Errorf()`.

Reply:
can we rename path to `/api/v1/config/reload` before release?

Reply:
maybe keep it as Prometheus format is better since users may want to  switch to vmagent  without  any concern. 
Or maybe expose two apis and delete the Prometheus format one when it's ready?

Reply:
Let's leave `/-/reload` handler in order to be compatible with Prometheus. @tenmozes ,  see https://www.robustperception.io/reloading-prometheus-configuration .